app/domain/authentication/authenticator_input.rb

```diff
@@ -6,6 +6,7 @@ module Authentication
 
   # :reek:InstanceVariableAssumption
   class AuthenticatorInput < ::Dry::Struct
+    constructor_type :schema
```

Comment:
what's this? why is it needed to be added?

Reply:
In order to let the "optional" values to be optionals. The default schema of struct doesn't allow optional keys. reference : https://dry-rb.org/gems/dry-struct/0.7/constructor-types/#constructor-types-were-removed-in-v0-5-0-use-code-transform_types-code-and-code-transform_keys-code-as-a-replacement

Reply:
i see, thanks